x-pack/plugins/reporting/export_types/png/server/lib/generate_png.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as Rx from 'rxjs';
+import { toArray, mergeMap } from 'rxjs/operators';
+import moment from 'moment';
+import { groupBy } from 'lodash';
+import { oncePerServer } from '../../../../server/lib/once_per_server';
+import { screenshotsObservableFactory } from './screenshots';
+import { getLayoutFactory } from './layouts';
+
+const getTimeRange = (urlScreenshots) => {
+  const grouped = groupBy(urlScreenshots.map(u => u.timeRange));
+  const values = Object.values(grouped);
+  if (values.length === 1) {
+    return values[0][0];
+  }
+
+  return null;
+};
+
+
+const formatDate = (date, timezone) => {
+  return moment.tz(date, timezone).format('llll');
+};
+
+function generatePngObservableFn(server) {
+  const screenshotsObservable = screenshotsObservableFactory(server);
+  const captureConcurrency = 1;
+  const getLayout = getLayoutFactory(server);
+
+  const urlScreenshotsObservable = (urls, headers, layout) => {
+    return Rx.from(urls).pipe(
+      mergeMap(url => screenshotsObservable(url, headers, layout),
+        (outer, inner) => inner,
+        captureConcurrency
+      )
+    );
+  };
+
+  const createPngWithScreenshots = async ({ title, browserTimezone, urlScreenshots }) => {
+    if (title) {
+      const timeRange = getTimeRange(urlScreenshots);
+      title += (timeRange) ? ` — ${formatDate(timeRange.from, browserTimezone)} to ${formatDate(timeRange.to, browserTimezone)}` : '';
+    }
+    return urlScreenshots[0].screenshots[0].base64EncodedData;
```

Comment:
timeRange and title unused.  We are assuming here only one screenshot, probably because we only support preview, not print layout.  Maybe worth asserting this and throwing an error if size !== 1.  Maybe with a comment that says if support for print layout is added, we can return multiple images, if it's not added, we can remove the array aspect of the parameter.